src/H5T.c

```diff
@@ -1504,9 +1504,9 @@ H5T_top_term_package(void)
                         if (H5DEBUG(T)) {
                             HDfprintf(H5DEBUG(T),
                                       "H5T: conversion function "
-                                      "0x%08lx failed to free private data for "
+                                      "0x%016zx failed to free private data for "
                                       "%s (ignored)\n",
-                                      (unsigned long)(path->conv.u.app_func), path->name);
+                                      (size_t)path->conv.u.app_func, path->name);
```

Comment:
Another place where an address is munged into a long, which is too small on Windows. This case is trickier since you can't really print function pointers portably, though. The best I can come up with is casting to size_t and printing that as hex.

Reply:
How much does portably printing matter here? Seems like using %p would be the most sane choice

Reply:
I would opt for %p.

Reply:
You can't use %p to print function pointers without warnings. It expects a void pointer and a function pointer is not a void pointer since it doesn't point to an object.

Reply:
Ah I'd overlooked the _function_ part. Agreed on the approach here then. I'd suggest casting to uintptr_t instead of size_t, but uintptr_t is an optional type.

Reply:
Unfortunately, uintptr_t was similarly cranky